cmd/prysmctl/withdrawal/withdrawal.go

```diff
@@ -0,0 +1,188 @@
+		"Please navigate to our website and make sure you understand the full implications of setting your withdrawal address. \n")
+	fmt.Println(au.Red("THIS ACTION WILL NOT BE REVERSIBLE ONCE INCLUDED. "))
+	fmt.Println(au.Red("You will NOT be able to change the address again once changed. "))
+
+	setWithdrawalAddressJsons := make([]*apimiddleware.SignedBLSToExecutionChangeJson, 0)
+	for _, foundFilePath := range foundFilePaths {
+		b, err := os.ReadFile(filepath.Clean(foundFilePath))
+		if err != nil {
+			return errors.Wrap(err, "failed to open file")
+		}
+		if string(b)[0:1] != "[" {
+			log.Warnf("provided file: %s, is not a list \n", foundFilePath)
+			continue
+		}
+		var to []*apimiddleware.SignedBLSToExecutionChangeJson
+		if err := json.Unmarshal(b, &to); err != nil {
+			return errors.Wrap(err, "failed to unmarshal file")
+		}
+		setWithdrawalAddressJsons = append(setWithdrawalAddressJsons, to...)
+	}
+	if len(setWithdrawalAddressJsons) == 0 {
+		return errors.New("the list of signed requests is empty")
+	}
+	if !c.Bool(SkipPromptsFlag.Name) {
+		for _, jsonOb := range setWithdrawalAddressJsons {
+			if err := verifyWithdrawalCertainty(r, jsonOb); err != nil {
+				return err
+			}
+		}
+	}
+	return callWithdrawalEndpoint(ctx, beaconNodeHost, setWithdrawalAddressJsons)
+}
+
+func verifyWithdrawalCertainty(r io.Reader, request *apimiddleware.SignedBLSToExecutionChangeJson) error {
+	au := aurora.NewAurora(true)
+	withdrawalConfirmation := request.Message.ToExecutionAddress
+	fmt.Println(au.Red("===================================="))
+	fmt.Println("YOU ARE ATTEMPTING TO CHANGE THE BLS WITHDRAWAL(" + au.Red(request.Message.FromBLSPubkey).String() + ") ADDRESS " +
+		"TO AN ETHEREUM ADDRESS(" + au.Red(request.Message.ToExecutionAddress).String() + ") FOR VALIDATOR INDEX(" + au.Red(request.Message.ValidatorIndex).String() + "). ")
+	_, err := withdrawalPrompt(withdrawalConfirmation, r)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func callWithdrawalEndpoint(ctx context.Context, host string, request []*apimiddleware.SignedBLSToExecutionChangeJson) error {
+	body, err := json.Marshal(request)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal json")
+	}
+	fullpath := host + apiPath
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, fullpath, bytes.NewBuffer(body))
+	if err != nil {
+		return errors.Wrap(err, "invalid format, failed to create new Post Request Object")
+	}
+	req.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("API request to %s responded with a status other than OK - status %v, body %v", fullpath, resp.Status, resp.Body)
+	}
+	log.Infof("Successfully published messages to update %d withdrawal addresses.", len(request))
+
+	log.Info("retrieving list of withdrawal messages known to node...")
```

Comment:
Is there a race condition here if one of the addresses was just included in a block? a bulletproof method would be to check the withdrawal credentials for those validators that fail to be in the pool.

Reply:
since the case is extremely edge involving proposing a block with committed withdrawal messages immediately after submission , we'll just print an additional message if we find any missing from the pool.